pkg/auth/token.go

```diff
@@ -347,3 +351,65 @@ func (v *TokenValidator) Middleware(next http.Handler) http.Handler {
 		next.ServeHTTP(w, r.WithContext(ctx))
 	})
 }
+
+// RFC9728AuthInfo represents the OAuth Protected Resource metadata as defined in RFC 9728
+type RFC9728AuthInfo struct {
+	Resource               string   `json:"resource"`
+	AuthorizationServers   []string `json:"authorization_servers"`
+	BearerMethodsSupported []string `json:"bearer_methods_supported"`
+	JWKSURI                string   `json:"jwks_uri"`
+	ScopesSupported        []string `json:"scopes_supported"`
+}
+
+// NewAuthInfoHandler creates an HTTP handler that returns RFC-9728 compliant OAuth Protected Resource metadata
+func NewAuthInfoHandler(issuer, jwksURL, resourceURL string, scopes []string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Set CORS headers for all requests
+		origin := r.Header.Get("Origin")
+		if origin == "" {
+			// Allow all origins if none specified. This should be fine because this is a discovery endpoint.
+			origin = "*"
+		}
+		w.Header().Set("Access-Control-Allow-Origin", origin)
+		w.Header().Set("Access-Control-Allow-Methods", "GET, OPTIONS")
+		// At least mcp-inspector requires these headers to be set for CORS. It seems to be a little
+		// off since this is a discovery endpoint, but let's make the inspector happy.
+		w.Header().Set("Access-Control-Allow-Headers", "mcp-protocol-version, Content-Type, Authorization")
+		w.Header().Set("Access-Control-Max-Age", "86400") // 24 hours
+
+		if r.Method == http.MethodOptions {
+			w.WriteHeader(http.StatusNoContent)
+			return
+		}
+
+		// if resourceURL is not set, return 404 as we shouldn't presume a resource URL
+		if resourceURL == "" {
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+
+		// Use provided scopes or default to 'openid'
+		supportedScopes := scopes
+		if len(supportedScopes) == 0 {
+			supportedScopes = []string{"openid"}
```

Comment:
The scopes seem again like something that might be configurable, but I didn't want to needlessly add more config options..